HousingFinanceInterimApi.Tests/V1/Infrastructure/DatabaseContext/LoadCashFileTests.cs

```diff
@@ -0,0 +1,183 @@
+            Assert.Equal(cashDump.Id, matchingCashLoad.UPCashDumpId);
+            Assert.Equal(DateTime.Today.Date, matchingCashLoad.Timestamp.Date);
+
+
+            _context.Entry(cashDump).Reload();
+            Assert.True(cashDump.IsRead);
+        }
+
+        /// <summary>
+        /// Given an invalid cash dump
+        ///     which has an unsuccessful filename
+        ///     or which is already read
+        /// Then does not create UPCashLoad
+        /// </summary>
+        /// <param name="rentAccount"></param>
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        public async void Given_An_Invalid_Cash_Dump__Does_Not_Create_UPCashLoad(bool fileNameIsSuccessful, bool cashDumpNotAlreadyRead)
+        {
+            // Arrange
+            var testClass = new UPCashLoadGateway(_context);
+
+            var cashDumpFileName = _fixture.Build<UPCashDumpFileName>()
+                .Without(x => x.Id)
+                .Without(x => x.Timestamp)
+                .With(x => x.IsSuccess, fileNameIsSuccessful)
+                .Create();
+            _context.UpCashDumpFileNames.Add(cashDumpFileName);
+            _context.SaveChanges();
+
+            var cashDump = _fixture.Build<UPCashDump>()
+                .Without(x => x.Id)
+                .With(x => x.UpCashDumpFileName, cashDumpFileName)
+                .With(x => x.FullText, DataGen.FullText())
+                .With(x => x.IsRead, !cashDumpNotAlreadyRead)
+                .Create();
+            _context.Add(cashDump);
+            _context.SaveChanges();
+
+            // Act
+            await testClass.LoadCashFiles().ConfigureAwait(false);
+
+            // Assert
+            var matchingCashLoad = _context.UpCashLoads.Where(x => x.UPCashDumpId == cashDump.Id).FirstOrDefault();
+            Assert.Null(matchingCashLoad);
+        }
+
+
+        [Theory]
+        [InlineData("invalid rent account")]
+        [InlineData("invalid payment source")]
+        public async void Given_A_UPCashDump_With_Invalid_FullText__ThrowsError(string fullText)
+        {
+            // Arrange
+            var testClass = new UPCashLoadGateway(_context);
+
+            var cashDumpFileName = _fixture.Create<UPCashDumpFileName>();
+            _context.UpCashDumpFileNames.Add(cashDumpFileName);
+            _context.SaveChanges();
+
+            var cashDump = _fixture.Build<UPCashDump>()
+                .Without(x => x.Id)
+                .With(x => x.UpCashDumpFileName, cashDumpFileName)
+                .With(x => x.FullText, fullText)
+                .Create();
+            _context.Add(cashDump);
+            _context.SaveChanges();
+
+            // Act
+            async Task Act() => await testClass.LoadCashFiles().ConfigureAwait(false);
+
+            // Assert
+            await Assert.ThrowsAsync<InvalidCashFileTextException>(Act).ConfigureAwait(false);
+
+        }
+    }
+
```

Comment:
newline

Reply:
Newline https://github.com/LBHackney-IT/housing-finance-interim-api/pull/149/commits/bce216cc33fd3c6ba4c6a2fab6f8aa14e8578cf6